create_data_image.sh

```diff
@@ -1,34 +1,28 @@
-#!/bin/bash
+#!/usr/bin/env bash
+
+set -euo pipefail
 
 # Currently everything runs as root, but that is going to change.
 USER_UID=0
 USER_GID=0
 
 source ./su_command.sh
 
-TOTAL_SIZE=0
-
 echo "Gathering applications..."
-APPS=
-for app in apps/*
-do
-	if test -f $app
-	then
-		APPS="${APPS} $app"
-		SIZE=$(stat -Lc %s ${app})
-		TOTAL_SIZE=$((${TOTAL_SIZE} + ${SIZE}))
-		echo "app: ${app} - $((${SIZE} / 1024)) kB"
-	elif [ $app != "apps/*" ]
-	then
-		echo "skipping non-regular file: $app"
-	fi
-done
-echo "Total data size: ${TOTAL_SIZE} bytes"
+mapfile -d $'\0' APPS < <(find apps/ -maxdepth 1 -type f -name '*.opk' -print0)
+if (( ${#APPS[@]} )); then
+	echo "${#APPS[@]} applications"
+	du -Lhc "${APPS[@]}"
+	APPS_SIZE="$(du -Lbc apps/*.opk | tail -1 | cut -f1)"
+else
+	echo "No application found in apps/*.opk"
+	APPS_SIZE=0
+fi
 echo
 
 # Pick a partition size that is large enough to contain all files but not much
 # larger so the image stays small.
-IMAGE_SIZE=$((8 + ${TOTAL_SIZE} / (920*1024)))
+IMAGE_SIZE=$((8 + APPS_SIZE / (920*1024)))
```

Comment:
Shouldn't this be $APPS_SIZE?

Reply:
This is bash, there is no need to add `$` inside of math expressions